doc/documentation/routing.rst

```diff
@@ -7,89 +7,55 @@ handled differently on Windows and Linux and is explained for both
 operating systems in the sections below.
 
 To identify each side of a route we will use the terms *client* and
-*target*. The *client* is your computer where pyads runs on. The
-*target* is you plc or remote computer which you want to connect to.
-
-Creating routes on Windows
---------------------------
-
-On Windows you don’t need to manually add the routes with pyads but
-instead you use the TwinCAT Router UI ( TcSystemManager) which comes
-with the TwinCAT installation. Have a look at the TwinCAT documentation
-on `infosys.beckhoff.com <https://infosys.beckhoff.de/english.php?content=../content/1033/TcSystemManager/Basics/TcSysMgr_AddRouteDialog.htm&id=>`__
+*target*. The *client* is your computer where pyads runs. The
+*target* is your plc or remote computer which you want to connect to.
+
+For the communication to work there needs to be routes in both directions, both
+from the client to the target and from the target to the client. The route from
+the client to the target is created when a Connection is created, but the route
+from the target to the client needs to be created manually according to
+instructions below.
+
+Creating target to client route(s) on Windows
+---------------------------------------------
+
+On Windows you can't add the routes from the target to the client with pyads,
+instead you use the TwinCAT Router UI (TcSystemManager) which comes with the
+TwinCAT installation. Have a look at the TwinCAT documentation on
+`infosys.beckhoff.com (TwinCAT2) <https://infosys.beckhoff.de/english.php?content=../content/1033/TcSystemManager/Basics/TcSysMgr_AddRouteDialog.htm&id=>`__
+/ `infosys.beckhoff.com (TwinCAT3) <https://infosys.beckhoff.com/english.php?content=../content/1033/tc3_system/html/tcsysmgr_addroutedialog.htm&id=>`__
 for further details.
 
-Creating routes on Linux
-------------------------
-
-To create a new route on Linux you can simply use the Connection class.
-It connects to the target and creates a route to it on your client.
-
-.. code:: python
-
-   >>> import pyads
-   >>> remote_ip = '192.168.0.100'
-   >>> remote_ads = '5.12.82.20.1.1'
-   >>> with pyads.Connection(remote_ads, pyads.PORT_TC3PLC1, remote_ip) as plc:
-   >>>     plc.read_by_name('.TAG_NAME', pyads.PLCTYPE_INT)
-
-.. note::
-
-  You still need to create a route from the target to the client. You
-  can do this manually on your target or you can use the function
-  :py:func:`.add_route_to_plc` as explained below!
-
-Get the AMS address of the local machine. This may need to be added to
-the routing table of the remote machine.
-
 .. note::
 
-  On Linux machines at least one route must be added before the
-  call to :py:func:`.get_local_address` will function properly.
-
-Optionally, a local AmsNetId can be manually set before adding a route.
```

Comment:
We shouldn't skip this part. Setting local netid is necessary in some cases.

Reply:
Interesting. Since I'm still a beginner of PLC programming I've had a few struggles recently. A small background: I have a python program that worked on Windows where I set up the Connection like this:
plc = pyads.Connection(self.target_ams_net_id, pyads.PORT_TC3PLC1)  # Works great on Windows, but not on Linux
To get it working in Linux I had to do add target ip:
plc = pyads.Connection(self.target_ams_net_id, pyads.PORT_TC3PLC1, self.target_ip)  # Works on both Linux and Windows
The difference is that on Linux I call the add_route_to_plc, which I don't on Windows.

But my question is probably: So there are situations where you do calls which require local netid but you have not set up a Connection?

Reply:
Yes, if you want to change your local netid you need to do this before you create your Connection object.

![grafik](https://user-images.githubusercontent.com/4641542/140637255-c4de3027-1277-42aa-91d3-3d02c0ff9db8.png)